keps/sig-storage/1432-volume-health-monitor/README.md

```diff
@@ -644,38 +680,260 @@ _This section must be completed when targeting alpha to a release._
+  Enabling the feature from the controller side will allow events to be reported on PVCs when
   abnormal volume conditions are detected.
 
 * **Can the feature be disabled once it has been enabled (i.e. can we rollback
   the enablement)?**
-  Yes. Uninstalling the health monitoring controller and agent sidecars will
-  disable the feature. Existing events will not be removed but they will
-  disappear after a period of time. Disabling the feature should not break
-  an existing application as these events are for humans only, not for automation. 
+  Yes. Uninstalling the health monitoring controller sidecar will disable the feature from
+  the controller side.
+  Disabling the feature gate on Kubelet will prevent Kubelet from monitoring volume health.
+  Existing events will not be removed but they will disappear after a period of time.
+  Disabling the feature should not break an existing application as these events are for humans
+  only, not for automation.
 
 * **What happens if we reenable the feature if it was previously rolled back?**
   Events will be added to PVCs or Pods when abnormal volume conditions are
   detected again.
 
 * **Are there any tests for feature enablement/disablement?**
-  Since there is no feature gate for this feature and the only way to enable or
-  disable this feature is to install or unistall the sidecars, we cannot write
-  tests for feature enablement/disablement.
+  There will be unit tests for the feature `VolumeHealth` enablement/disablement.
+  Since there is no feature gate for this feature on the controller side and the only way to
+  enable or disable this feature is to install or unistall the sidecar, we cannot write
+  tests for feature enablement/disablement on the controller side.
+
+### Rollout, Upgrade and Rollback Planning
+
+_This section must be completed when targeting beta graduation to a release._
+
+* **How can a rollout fail? Can it impact already running workloads?**
+  Try to be as paranoid as possible - e.g., what if some components will restart
+   mid-rollout?
+   This feature does not have a feature gate on the controller side. It is enabled when
+   the health monitoring controller sidecar is deployed with the CSI driver.
+   So the only way for a rollout to fail is that deploying the health
+   monitoring controller sidecar with the CSI driver fails. If
+   the health monitoring controller cannot be deployed, no events on volume
+   condition will be reported on PVCs.
+
+   If enabling the `VolumeHealth` feature fails, no event on volume condition will be
+   reported on the pod.
+
+* **What specific metrics should inform a rollback?**
+  An event will be recorded on the PVC when the controller has successfully retrieved an
+  abnormal volume condition from the storage system. When other errors occur in the controller,
+  the errors will also be recorded as events.
+
+  In Kubelet, an event will be recorded on the Pod when Kubelet has successfully retrieved an
```

Comment:
We don't have kubelet metrics on the csi call itself. This is something that @Jiawei0227 was considering adding in general.